### PR TITLE
Added Slideshow feature

### PR DESCRIPTION
**Pull request by tanujbordikar**
Solved the Issue no. #72 

**JS Implementation:** Incorporate necessary JavaScript to make the slideshow interactive and dynamic.
**CSS Styling:** Apply responsive styling using the BEM (Block, Element, Modifier) methodology to ensure the slideshow adapts to various screen sizes.

https://github.com/dear-digital/shopify-theme-skeleton/assets/91179998/b60e93bf-1cd4-4422-bb3f-ca86c4f38e36

### CODE REVIEW

File: sections/slideshow.liquid [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    </span>
+    <span id="next" class="slider-button next">
+      <svg class="slider__svg-button next" viewBox="0 0 100 100">
+        <path style="fill:{{text_color}};" d="M 10,50 L 60,100 L 70,90 L 30,50  L 70,10 L 60,0 Z" class="arrow" transform="translate(100, 100) rotate(180) "></path>
+      </svg>
+    </span>
+  </div>
+</div>
+
+<script src="{{ 'slideshow.js' | asset_url }}" defer="defer"></script>
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:36]: slideshow should work without saving, checkout the [theme editor events](https://shopify.dev/docs/themes/architecture/sections/integrate-sections-with-the-theme-editor) for re-running the scripts when the section in editor changes

File: sections/slideshow.liquid [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    <span id="slideCurrent"></span> / <span id="slideTotal"></span>
+    <span id="prev" class="slider-button prev">
+      <svg class="slider__svg-button prev" viewBox="0 0 100 100">
+        <path style="fill:{{text_color}};" d="M 10,50 L 60,100 L 70,90 L 30,50  L 70,10 L 60,0 Z" class="arrow"></path>
+      </svg>
+    </span>
+    <span id="next" class="slider-button next">
+      <svg class="slider__svg-button next" viewBox="0 0 100 100">
+        <path style="fill:{{text_color}};" d="M 10,50 L 60,100 L 70,90 L 30,50  L 70,10 L 60,0 Z" class="arrow" transform="translate(100, 100) rotate(180) "></path>
+      </svg>
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:31]: svg's can come from snippets, you can also pass classes as parameters inside render liquid tag

File: sections/slideshow.liquid [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+<div id="sliderContainer" class="slider-container" style="color:{{ text_color }};">
+  {% for block in section.blocks %}
+    <div class="hero-slide" style="background-image:url({{ block.settings.slide_bg | img_url: 'original' }});">
+      <div class="page-width hero-slide-content">
+        <h2>{{ block.settings.main_title }}</h2>
+        <p style="color:{{ text_color }};">{{ block.settings.sub_title }}</p>
+        <a
+          class="btn"
+          style="color:{{ text_color }};border-color:{{ text_color }};"
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:12]: avoid all inline styles, instead use section.id in liquid to identify the selector and use it in style tag refer sections in dawn theme. If using tailwind create seperate schemas for dynamically adding tailwind classes refer the demo section.

File: sections/slideshow.liquid [hunk 4]
```diff
@@ -0,0 +1,97 @@
+{% assign text_color = section.settings.text_color %}
+{{ 'slideshow.css' | asset_url | stylesheet_tag }}
+
+<div id="sliderContainer" class="slider-container" style="color:{{ text_color }};">
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:4]: Use appropriate BEM style classes everywhere in the section

**Review by mihir-bombay-studio — CHANGES_REQUESTED** [review]
Can you redo the following changes? Thanks!

File: sections/slideshow.liquid [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    </span>
+    <span id="next" class="slider-button next">
+      {% render 'icon-right-arrow' %}
+    </span>
+  </div>
+</div>
+
+{%- if request.design_mode -%}
+  <script src="{{ 'slideshow.js' | asset_url }}" defer="defer"></script>
+{% endif %}
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:32]: slideshow will only work in theme editor, not on live website with this condition.

File: sections/slideshow.liquid [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+<script>
+  document.addEventListener('DOMContentLoaded', function () {
+    document.addEventListener('shopify:section:load', function (event) {
+      if (event.detail.sectionId === '{{ section.id }}') {
+      }
+    });
+  });
+</script>
```
Comment (mihir-bombay-studio) [inline sections/slideshow.liquid:41]: blank script

**Review by mihir-bombay-studio — CHANGES_REQUESTED** [review]
@tanujbordikar prepend all file names /assets and /sections with dev__.
Review the entire code once, compare it with Dawn slideshow section, use chatgpt to optimise and correct the code.

**Comment by tanujbordikar** [comment]
Sir can you provide more information about the re-running the scripts when the section in editor changes. I have tried to solve it but couldn't able to achieve what you have asked for!

**Pull request closed without merging**